ibazel/output_runner/output_runner.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package output_runner
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"strconv"
+
+	"github.com/bazelbuild/bazel-watcher/ibazel/workspace_finder"
+	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
+)
+
+var runOutput = flag.Bool(
+	"run_output",
+	false,
+	"Search for commands in Bazel output that match a regex and execute them, the default path of file should be in the workspace root .bazel_fix_commands.json")
+var runOutputInteractive = flag.Bool(
+	"run_output_interactive",
+	true,
+	"Use an interactive prompt when executing commands in Bazel output")
+
+type OutputRunner struct{}
+
+type Optcmd struct {
+	Regex string `json:"regex"`
+	Command string `json:"command"`
+	Args []string `json:"args"`
+}
+
+func New() *OutputRunner {
+	i := &OutputRunner{}
+	return i
+}
+
+func (i *OutputRunner) Initialize(info *map[string]string) {}
+
+func (i *OutputRunner) TargetDecider(rule *blaze_query.Rule) {}
+
+func (i *OutputRunner) ChangeDetected(targets []string, changeType string, change string) {}
+
+func (i *OutputRunner) BeforeCommand(targets []string, command string) {}
+
+func (i *OutputRunner) AfterCommand(targets []string, command string, success bool, output *bytes.Buffer) {
+	if *runOutput == false || output == nil {
+		return
+	}
+
+	jsonCommandPath := ".bazel_fix_commands.json"
+	defaultRegex := Optcmd{
+		Regex: "^(buildozer) '(.*)'\\s+(.*)$",
```

Comment:
Just as a note this is a regexp, as opposed to a pushdown automata, and so complex stuff like escaping single quotes is not going to work.There is some subtlty to correctly parsing bash that I think is elided in here. For example `buildozer "do stuff" //my:target` and `buildozer print //target` are not going to workeven though they are valid. I don't think it's worth the effort to construct a parser for this, but I think it would be worth indicating to rules authors in a comment.

Reply:
We are not parsing bash syntax at all. We are just parsing the build command output and constructing our own shell command from the captured groups. The output could just as easily be a compile or lint error with file names and line numbers.

Reply:
I think we are not parsing arbitrary bash here. Rules authors may design specific command format, and since JSON is configurable, users can parse the commands they are looking for. Unless we change the buildozer commands, this default regex will work. Am I right? Also, where is a good place to leave a comment?